includes/event/event-repository.php

```diff
@@ -96,6 +97,8 @@ public function trash_event( Event $event ) {
 		if ( ! $result ) {
 			return false;
 		}
+		unset( $this->cache[ $event->id() ] );
```

Comment:
I think we shouldn't add caching here, because there's a version of the repository that is specific for caching (`Cached_Event_Repository`), so adding caching in the non-cached repository feels a bit weird to me. In addition, as I mention in another comment, I think that if we pass only the data that the block requires as an attribute, instead of passing the event id, there will be no need for caching.

(The `Cached_Event_Repository` might not have been the greatest idea, but given it currently exists, I think we should first re-evaluate whether we should remove it before adding caching directly in this repository).

Reply:
I think it might be helpful to remove the caching from this PR and create an issue where we decide on the best approach.